src/script/lua_api/l_mainmenu.cpp

```diff
@@ -451,7 +451,8 @@ int ModApiMainMenu::l_check_mod_configuration(lua_State *L)
 		spec.path = modpath;
 		spec.virtual_path = virtual_path;
 		if (!parseModContents(spec)) {
-			throw LuaError("Not a mod!");
+			warningstream << "Mod \"" << spec.name
```

Comment:
I think it would be better to return `nil`, error message here (push nil, then the error string, then `return 2;`). Then the Lua side of things can handle the error (without `pcall`), e.g by showing an "error" icon in the select mods dialog.

Or just use a `pcall` on the Lua side of things.

I don't think it's a good idea for the main menu to only complain in logs when a mod is straight up incomplete. We complain when a mod is missing dependencies as well.

Reply:
I made correct error handling without terminating anything. Now, Lua side now knows about `parseModContents()` result for each mod via `mod.valid` field and displays content red with error icon if it is invalid. I also added error message as without it error reason is not obvious.

I'm not sure is my realization is totally correct, but i tried to make enough extendable error handling.